### PR TITLE
ci: Start separate registries for gke builds

### PR DESCRIPTION
**Pull request by nebril**

### CODE REVIEW

**Comment by nebril** [comment]
test-gke

**Comment by nebril** [comment]
test-gke

**Comment by nebril** [comment]
test-gke

**Comment by nebril** [comment]
test-gke

**Comment by nebril** [comment]
test-gke

**Comment by nebril** [comment]
test-gke

**Comment by nebril** [comment]
test-4.9

File: test/gke/stop-registry.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+set -e
+
+script_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+docker kill "$(cat ${script_dir}/registry_container)"
```
Comment (aanm) [inline test/gke/stop-registry.sh]: `docker kill -f`?

**Review by aanm — COMMENTED** [review]
Minor suggestion

File: test/gke/stop-registry.sh (hunk 1 above)
Comment (nebril) [reply test/gke/stop-registry.sh]: There is not `-f` flag for `docker kill` :(.

File: test/gke/start-registry.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+docker run -d \
+	--restart=always \
+	-v /opt/registry/certs:/certs \
+	-e REGISTRY_HTTP_ADDR=0.0.0.0:443 \
+	-e REGISTRY_HTTP_TLS_CERTIFICATE=/certs/domain.crt \
+	-e REGISTRY_HTTP_TLS_KEY=/certs/domain.key \
+	-p 443 \
+	registry:2 > ${script_dir}/registry_container
+
+docker inspect "$(cat registry_container)" | jq -r '.[0]["NetworkSettings"]["Ports"]["443/tcp"][0]["HostPort"]' > ${script_dir}/registry_port
```
Comment (brb) [inline test/gke/start-registry.sh]: For the sake of consistency, maybe `s:cat registry_container:cat $script_dir/registry_container`?

File: test/gke/stop-registry.sh (hunk 1 above)
Comment (brb) [reply test/gke/stop-registry.sh]: You could `docker rm -f $(cat .../registry_container)`.

**Review by brb — APPROVED** [review]
LGTM, just a few nits. Also, it'd be useful two add a couple of sentences into the commit message explaining why the separate registry is needed for GKE builds.

**Comment by nebril** [comment]
test-gke

**Comment by nebril** [comment]
Added a small locking library for bash scripts so that no two jobs change docker configuration in parallel.

**Review by tklauser — APPROVED** [review]

**Comment by tklauser** [comment]
Tests passed on GKE which this PR is targeting, thus merging.

**Pull request merged**